xrpl/models/flags.py

```diff
@@ -1,110 +1,74 @@
-        "tf_one_asset_lp_token": 0x00200000,
-        "tf_limit_lp_token": 0x00400000,
-    },
-    "NFTokenCreateOffer": {
-        "tf_sell_token": 0x00000001,
-    },
-    "NFTokenMint": {
-        "tf_burnable": 0x00000001,
-        "tf_only_xrp": 0x00000002,
-        "tf_trustline": 0x00000004,
-        "tf_transferable": 0x00000008,
-    },
-    "OfferCreate": {
-        "tf_passive": 0x00010000,
-        "tf_immediate_or_cancel": 0x00020000,
-        "tf_fill_or_kill": 0x00040000,
-        "tf_sell": 0x00080000,
-    },
-    "Payment": {
-        "tf_no_direct_ripple": 0x00010000,
-        "tf_partial_payment": 0x00020000,
-        "tf_limit_quality": 0x00040000,
-    },
-    "PaymentChannelClaim": {
-        "tf_renew": 0x00010000,
-        "tf_close": 0x00020000,
-    },
-    "TrustSet": {
-        "tf_set_auth": 0x00010000,
-        "tf_set_no_ripple": 0x00020000,
-        "tf_clear_no_ripple": 0x00040000,
-        "tf_set_freeze": 0x00100000,
-        "tf_clear_freeze": 0x00200000,
-    },
-    "EnableAmendment": {
-        "tf_got_majority": 0x00010000,
-        "tf_lost_majority": 0x00020000,
-    },
-}
-
 
 class FlagInterface(TypedDict):
     """A TypedDict to define transaction flags by bool."""
 
     pass
 
 
+def _flag_enum_to_dict(flag_enum: Type[Enum]) -> Dict[str, int]:
+    return {flag.name.lower(): flag.value for flag in flag_enum}
+
+
 def interface_to_flag_list(
     tx_type: Union[TransactionType, PseudoTransactionType],
     tx_flags: Dict[str, bool],
 ) -> List[int]:
     """Parse a list of flags expressed as integers from the FlagInterface.
 
     Args:
-        tx_type (Union[TransactionType, PseudoTransactionType]):
-            Type of the transaction.
-        tx_flags (dict):
-            FlagInterface
+        tx_type: Type of the transaction.
+        tx_flags: FlagInterface
 
     Returns:
         List[int]:
             A list of flags expressed as integers.
     """
-    if tx_type not in TX_FLAGS:  # if transaction types has no flags
+    from xrpl.models import transactions  # Avoid circular dependencies
+
+    # Get all exported classes with names of the form `{TxType}Flag`
+    # These are all the flag value enums for transactions/pseudo-transactions
+    flag_enums = [
+        f for f in transactions.__all__ if f.endswith("Flag") and "Asf" not in f
+    ]
+    pseudo_tx_flag_enums = [
+        f for f in transactions.pseudo_transactions.__all__ if f.endswith("Flag")
+    ]
+
+    # Key is transaction type name, value is mapping of flag name to int value
+    all_tx_flags: Dict[str, Dict[str, int]] = {
+        # The `:-4` here removes the `Flag` at the end of the class type to just get
+        # the transaction type name
+        **{f[:-4]: _flag_enum_to_dict(getattr(transactions, f)) for f in flag_enums},
+        **{
+            f[:-4]: _flag_enum_to_dict(getattr(transactions.pseudo_transactions, f))
+            for f in pseudo_tx_flag_enums
+        },
+    }
```

Comment:
```suggestion
    tx_flag_enums = [
        f for f in transactions.__all__ if f.endswith("Flag") and "Asf" not in f
    ]
    pseudo_tx_flag_enums = [
        f for f in transactions.pseudo_transactions.__all__ if f.endswith("Flag")
    ]

    # Key is transaction type name, value is mapping of flag name to int value
    all_tx_flags: Dict[str, Dict[str, int]] = {
        # The `:-4` here removes the `Flag` at the end of the class type to just get
        # the transaction type name
        **{f[:-4]: _flag_enum_to_dict(getattr(transactions, f)) for f in tx_flag_enums},
        **{
            f[:-4]: _flag_enum_to_dict(getattr(transactions.pseudo_transactions, f))
            for f in pseudo_tx_flag_enums
        },
    }
```
Renaming `flag_enums` to `tx_flag_enums` makes it specific.